fuel-crypto/src/secp256r1.rs

```diff
@@ -0,0 +1,175 @@
+//! secp256r1 (P-256) functions
+
+use crate::{
+    Error,
+    Message,
+};
+use ecdsa::RecoveryId;
+use fuel_types::Bytes64;
+use p256::ecdsa::{
+    Signature,
+    SigningKey,
+    VerifyingKey,
+};
+
+/// Combines recovery id with the signature bytes. See the following link for explanation.
+/// https://github.com/FuelLabs/fuel-specs/blob/master/src/protocol/cryptographic_primitives.md#public-key-cryptography
+fn encode_signature(signature: Signature, recovery_id: RecoveryId) -> [u8; 64] {
+    let mut signature: [u8; 64] = signature.to_bytes().into();
+    debug_assert!(signature[32] >> 7 == 0);
+    assert!(!recovery_id.is_x_reduced());
+
+    let v = recovery_id.is_y_odd() as u8;
+
+    signature[32] = (v << 7) | (signature[32] & 0x7f);
+    signature
+}
+
+/// Separates recovery id from the signature bytes. See the following link for
+/// explanation. https://github.com/FuelLabs/fuel-specs/blob/master/src/protocol/cryptographic_primitives.md#public-key-cryptography
+fn decode_signature(mut signature: [u8; 64]) -> Option<(Signature, RecoveryId)> {
+    let v = (signature[32] & 0x80) != 0;
+    signature[32] &= 0x7f;
+
+    let signature = Signature::from_slice(&signature).ok()?;
+
+    Some((signature, RecoveryId::new(v, false)))
+}
+
+/// Sign a prehashed message. With the given key.
+pub fn sign_prehashed(
+    signing_key: &SigningKey,
+    message: &Message,
+) -> Result<Bytes64, Error> {
+    let (signature, _) = signing_key
+        .sign_prehash_recoverable(&**message)
+        .map_err(|_| Error::FailedToSign)?;
+
+    let signature = signature.normalize_s().unwrap_or(signature);
+
+    // TODO: this is a hack to get the recovery id. The signature should be normalized
```

Comment:
Yep, I can make a PR to support that. However, it's not really blocking this PR since the version used here works, and the VM doesn't use this function so there is not performance penaly for now. I'm simply uncertain whether we're going to end up using this library for the signatures. If we end up adding more EC opcodes as proposed [in the spec issue](https://github.com/FuelLabs/fuel-specs/pull/484#pullrequestreview-1479624156), we're likely going to write our own EC library anyway.